Spell checking indicator shows when an inline prediction candidate for a proper noun is being shown

rr-codes opened the pull request:
#### 8521823072e7874f26ca06f742185dae605316d0
<pre>
Spell checking indicator shows when an inline prediction candidate for a proper noun is being shown
<a href="https://bugs.webkit.org/show_bug.cgi?id=258643">https://bugs.webkit.org/show_bug.cgi?id=258643</a>
rdar://110196591

Reviewed by Wenson Hsieh.

Use the `m_isHandlingAcceptedCandidate` flag to indicate when an inline prediction candidate is
being offered.

Setting text via `setMarkedText` eventually gets into `Editor::markAllMisspellingsAndBadGrammarInRanges`,
so exit early in that method if the flag is set so that spelling and grammer checking does not get
erroneously applied.

* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::setComposition):
(WebCore::Editor::markAllMisspellingsAndBadGrammarInRanges):
* Tools/TestWebKitAPI/Tests/mac/WKWebViewMacEditingTests.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/265725@main">https://commits.webkit.org/265725@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/969e4662964d53aae2d27bd129c9e91984e5a955

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11669 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11869 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12236 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13311 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11114 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11688 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14257 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11849 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13984 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11833 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12677 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9899 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13734 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9969 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10595 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17728 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11045 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10749 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13924 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11146 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9195 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10323 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2821 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14603 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11003 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/TestWebKitAPI/Tests/mac/WKWebViewMacEditingTests.mm:
```diff
@@ -178,4 +180,42 @@ - (SlowTextInputContext *)_web_superInputContext
+    auto string = adoptNS([[NSMutableAttributedString alloc] init]);
+    [string appendAttributedString:[[NSAttributedString alloc] initWithString:@"wedn"]];
+    [string appendAttributedString:[[NSAttributedString alloc] initWithString:@"esday" attributes:@{
+        NSForegroundColorAttributeName : NSColor.grayColor
```
Nit - `adoptNS` or `-autorelease` to fix these leaked `NSAttributedString`s.

rr-codes committed:
Spell checking indicator shows when an inline prediction candidate for a proper noun is being shown
https://bugs.webkit.org/show_bug.cgi?id=258643
rdar://110196591

Reviewed by Wenson Hsieh.

Use the `m_isHandlingAcceptedCandidate` flag to indicate when an inline prediction candidate is
being offered.

Setting text via `setMarkedText` eventually gets into `Editor::markAllMisspellingsAndBadGrammarInRanges`,
so exit early in that method if the flag is set so that spelling and grammer checking does not get
erroneously applied.

* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::setComposition):
(WebCore::Editor::markAllMisspellingsAndBadGrammarInRanges):
* Tools/TestWebKitAPI/Tests/mac/WKWebViewMacEditingTests.mm:
(TEST):

Canonical link: https://commits.webkit.org/265725@main
